modules/editorial-metadata/editorial-metadata.php

```diff
@@ -332,6 +375,10 @@ public function update_editorial_metadata_term( $term_id, $args ) {
 	 * @return bool $result Whether or not the term was deleted
 	 */
 	public function delete_editorial_metadata_term( $term_id ) {
+		$term = $this->get_editorial_metadata_term_by( 'id', $term_id );
+		$post_meta_key = $this->get_postmeta_key( $term_id, $term->type );
+		delete_post_meta_by_key( $post_meta_key );
```

Comment:
This ensures all the keys are gone, before we delete the term. Not checking the result because if a post doesn't use it it will give back false. That's expected technically